src/Symfony/Component/Console/Command/DumpCompletionCommand.php

```diff
@@ -0,0 +1,111 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Console\Command;
+
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\ConsoleOutputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Process\Process;
+
+/**
+ * Dumps the completion script for the current shell.
+ *
+ * @author Wouter de Jong <wouter@wouterj.nl>
+ */
+class DumpCompletionCommand extends Command
+{
+    protected function configure()
+    {
+        $fullCommand = $_SERVER['PHP_SELF'];
+        $commandName = basename($fullCommand);
+        $fullCommand = realpath($fullCommand) ?: $fullCommand;
+
+        $this
+            ->setName('completion')
+            ->setDescription('Dump the shell completion script')
+            ->setHelp(<<<EOH
+The <info>%command.name%</info> command dumps the shell completion script required
+to use shell autocompletion.
+
+<comment>Static installation
+-------------------</comment>
+
+Dump the script to a global completion file and restart your shell:
+
+    <info>%command.full_name% bash | sudo tee /etc/bash_completion.d/${commandName}</info>
+
+Or dump the script to a local file and source it:
+
+    <info>%command.full_name% bash > completion.sh</info>
+
+    <comment># source the file whenever you use the project</comment>
+    <info>source completion.sh</info>
+
+    <comment># or add this line at the end of your "~/.bashrc" file:</comment>
+    <info>source /path/to/completion.sh</info>
+
+<comment>Dynamic installation
+--------------------</comment>
+
+Add this add the end of your shell configuration file (e.g. <info>"~/.bashrc"</info>):
```

Comment:
It means the script is generated every time you open a new shell.

The disadvantage is a slight loss of performance when opening the new shell. The advantage is always having the up to date completion script.

Reply:
But which copy of console do you point it to?

Speaking of, the completion script should catch the exception where the completion command is unrecognized.

Reply:
It shouldn't be put into the `.bashrc` What about multiple symfony projects ? Couldn't we have like a dump inside the project itself and then a bash script in the bashrc that check for the existence of this file (maxdepth=2 for example) or even an env variable that contains the target to source ?

One last thing : I'm not completely sure how it will work yet but it will require php to autocomplete or just to dump the completion script ? If it requires php to autocomplete won't that kill DX when using docker for example ?